plugins/modules/win_psmodule.ps1

```diff
@@ -460,13 +466,13 @@ if ($repo_user -and $repo_pass ) {
 }
 
 if ( ($allow_clobber -or $allow_prerelease -or $skip_publisher_check -or
-        $required_version -or $minimum_version -or $maximum_version) ) {
+        $required_version -or $minimum_version -or $maximum_version -or $accept_license) ) {
     # Update the PowerShellGet and PackageManagement modules.
     # It's required to support AllowClobber, AllowPrerelease parameters.
-    Install-PrereqModule -AllowClobber $allow_clobber -CheckMode $check_mode -AcceptLicense $accept_license
+    Install-PrereqModule -AllowClobber $allow_clobber -CheckMode $check_mode -AcceptLicense $accept_license -Repository $repo
```

Comment:
I was thinking about the repository issue yesterday and whether or not to add it, but I think it might be best not to explicitly specify it. If the repo you supply is private it probably doesn't have the pre-reqs, so this will fail in those cases. If we left it blank, it will search all repositories, and then find it in the public one unless PSGallery was removed.

For our tests, we should remove PSGallery to avoid pulling from the internet (and to avoid the conflict of it being available in both).

Reply:
I’m not sure, if someone specifies a repository then I would argue that it should be used for everything, even pre reqs. Even then the pre reqs are only installed when the newer arguments have been requested so I would be surprised if this actual code is run a lot in the wild.

Happy to talk about it further before a new release is made to ensure we cover all our bases.

Reply:
My best guess is that the code is run a lot in the wild, but only once for a given host, so it will affect mostly new hosts being brought up. With the repo change, if the first use of `win_psmodule` on the new host installs a module from PSGallery, there will be no issue:

```
# first time running on a new host, pre-reqs will be updated
- community.windows.win_psmodule:
    name: psprivilege
```

but if the first use installs from a private repo, the pre-req install will fail.

```
# first time running on a new host, pre-reqs can't be found
- community.windows.win_psmodule:
    name: ACMEModule
    repo: PSACME
```